lib/manageiq/providers/ibm_cloud/cloud_tools/vpc_sdk/instance_actions.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module ManageIQ
+  module Providers
+    module IbmCloud
+      module CloudTools
+        module VpcSdk
+          class InstanceActions < ManageIQ::Providers::IbmCloud::CloudTools::Base::Instance
```

Comment:
Hey @jaredm-ibm is the reason you're subclassing `Base::Instance` here so that it can share some of those methods?  It is confusing for `InstanceActions` and `Instance` and `Instances` to all inherit from `Instance`.  At least `InstanceActions` isn't `kind_of?(Instance)`

If you're trying to pull e.g. `wait_for` into each of these maybe a module/mixin would be a better choice?

Reply:
The reason for the subclass of Base::Instance is to have standard @parent and logger method. The wait_for is just there because it is generic enough to be useful elsewhere.

Reply:
:+1: that's where we were saying it'd be better to use a mixin if you're trying to share common code.

> So, some strange things I see immediately are things like Instances is a CloudTools::Base::Instance and InstanceActions < CloudTools::Base::Instance. In these cases, it seems you really just want to share the helper methods wait_for and logger, so for those the general approach is to use a module mixin.

Inheritance implies the child class is a type of the parent class /cc @Fryguy

Reply:
Changed the naming to SDK::Branch for SDK Wrappers that branch off of CloudTool and SDK::Leaf for Classes that deal with specific aspects of the SDK Branch. This gives you your 'is a' relationship without breaking my design.